java/java.hints/src/org/netbeans/modules/java/hints/introduce/IntroduceMethodPanel.java

```diff
@@ -254,6 +253,7 @@ private Preferences getPreferences() {
     
     public void setOkButton( JButton btn ) {
         this.btnOk = btn;
+        changeSupport.run();
```

Comment:
On other places, ChangeSupport.run is run off EDT, since validators tend to use Parsing API which might be quite slow.

Reply:
it is quite fast even in large files but I agree in principle. It is also run on `focusLost()` etc, so this would require more work.

Reply:
so i debugged this and the whole panel initialization runs off-EDT(!), this call included ;)

What I did though to not make it worse: I removed this call to `run()` again, and triggered validation when the default value is set.

Reply:
well. this creates another problem i wanted to avoid. Now I probably going to have to init `valid` with true.